spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -148,8 +149,9 @@ protected Statistics estimateStatistics(Snapshot snapshot) {
 
     for (CombinedScanTask task : tasks()) {
       for (FileScanTask file : task.files()) {
-        // TODO: if possible, take deletes also into consideration.
-        double fractionOfFileScanned = ((double) file.length()) / file.file().fileSizeInBytes();
+        // take deletes also into consideration.
+        long deletesSizeInBytes = file.deletes().stream().mapToLong(DeleteFile::fileSizeInBytes).sum();
+        double fractionOfFileScanned = ((double) file.length()) / (file.file().fileSizeInBytes() + deletesSizeInBytes);
```

Comment:
Thanks @felixYyu for your contribution.

As per my understanding a single delete file can be associated with multiple data files, (also position delete doesn't store complete deleted row info also (ref [this](https://iceberg.apache.org/spec/#position-delete-files)) there is also case of equality deletes (ref : [this](https://iceberg.apache.org/spec/#equality-delete-files)) where we just store the matching pred, hence it will not be so straight forward to estimate size / row count.

Hence row count post applying delete files will be hard to estimate without actually applying them, hence I left it as a todo.

From the looks of it, this approach looks incorrect to me (may be I am wrong here) will leave it to the better judgement of more experienced folks of iceberg.